includes/admin_import.php

```diff
@@ -160,23 +160,29 @@ function tsml_import_page()
         if (!empty($_POST['tsml_add_data_source']) && isset($_POST['tsml_nonce']) && wp_verify_nonce($_POST['tsml_nonce'], $tsml_nonce)) {
 
             //sanitize URL, name, parent region id, and Change Detection values
-            $data_source_url = trim(esc_url_raw($_POST['tsml_add_data_source'], array('http', 'https')));
+            $data_source_url = $imported_data_source_url = trim(esc_url_raw($_POST['tsml_add_data_source'], array('http', 'https')));
             $data_source_name = sanitize_text_field($_POST['tsml_add_data_source_name']);
             $data_source_parent_region_id = (int) $_POST['tsml_add_data_source_parent_region_id'];
             $data_source_change_detect = sanitize_text_field($_POST['tsml_add_data_source_change_detect']);
 
+            //use c4r sheets service for Google Sheets URLs
+            if (strpos($data_source_url, 'docs.google.com/spreadsheets') !== false) {
+                $url_parts = explode('/', $data_source_url);
+                $sheet_id = $url_parts[5];
+                $imported_data_source_url = 'https://sheets.code4recovery.org/tsml/' . $sheet_id;
+            }
```

Comment:
Struck me this `sheets.code4recovery.org/tsml/` should be a top level constant at some point.